firmware/application/apps/ui_scanner.cpp

```diff
@@ -72,453 +74,251 @@ void ScannerThread::run() {
-		for(auto& entry : database) {
-			// FIXME
-			if (entry.type == RANGE) {
+	size_t step = 0;
+	size_t	max_entry = 0 ;
+	size_t	def_step = 0 ;
+	size_t entry_full = 0 ;
+	std::string scanner_file ;
+	
+	switch (mod_type) {
+	case NFM: scanner_file = "SCANNER_NFM";	def_step = 12500;	add_children({&field_bw_NFM });				break;
+	case AM:  scanner_file = "SCANNER_AM";	def_step = 9000;	add_children({&field_bw_AM});				break;
+	case FM:  scanner_file = "SCANNER_FM";	def_step = 100000;	add_children({&field_bw_FM});				break;
+	case ANY: scanner_file = "SCANNER_ANY";	def_step = 10000;	add_children({&field_bw_ANY	});				break;
+	default: 																								break;
+}
+	// LEARN FREQUENCIES
+	
+	if ( load_freqman_file(scanner_file, database)  ) {
+		text_cycle1.set("SCANNER FILE PRESENT" );
+		for(auto& entry : database) {						// READ LINE PER LINE
+			
+			if ( (entry.type == RANGE)  && (entry_full == 0) )  {										//RANGE
 				
 				switch (entry.step) {
-				case STEP_DEF:	step_nfm = 12500  	;  	break ;				// default
-				case AM_US:		step_nfm = 10000  	;  	break ;
-				case AM_EUR:	step_nfm = 9000  	;  	break ;
-				case NFM_1: 	step_nfm = 12500  	;  	break ;
-				case NFM_2:    	step_nfm = 6250 	;	break ;	
-				case FM_1:		step_nfm = 100000  	;  	break ;
-				case FM_2:		step_nfm = 50000  	; 	break ;
-				case N_1:		step_nfm = 25000  	;  	break ;
-				case N_2:		step_nfm = 250000  	;  	break ;
-				default:		step_nfm = 12500 	;							//error
+				case STEP_DEF:	step 	= def_step 	;  	break ;				// default
+				case AM_US:		step 	= 10000  	;  	break ;
+				case AM_EUR:	step 	= 9000  	;  	break ;
+				case NFM_1: 	step 	= 12500  	;  	break ;
+				case NFM_2:    	step	= 6250 		;	break ;	
+				case FM_1:		step 	= 100000  	;  	break ;
+				case FM_2:		step 	= 50000  	; 	break ;
+				case N_1:		step 	= 25000  	;  	break ;
+				case N_2:		step 	= 250000  	;  	break ;
+				case AIRBAND:	step 	= 8330  	;  	break ;
+				default:		step 	= def_step 	;						//error
 						}
-				
-				for (uint32_t i=entry.frequency_a; i < entry.frequency_b; i+= step_nfm) 	{
-					
-					max_entry_nfm = max_entry_nfm  + 1 ;
+			
+				for (uint32_t i=entry.frequency_a; i < entry.frequency_b; i+= step) 	{
+					max_entry = max_entry  + 1 ;
 
-					
-				if ( max_entry_nfm >= MAX_DB_ENTRY) { 
-					text_modulation.set( "Many entries:R: " + to_string_dec_uint(entry.frequency_a)  );break;}
+				if ( max_entry >= MAX_DB_ENTRY) 
+				{
+					DEBUG(3, "Too many R:" + to_string_dec_uint(max_entry) ); 
+					text_modulation.set( "Many entries:R: " + to_string_dec_uint(entry.frequency_a)  );
+					entry_full = 1;
+					break;
+				}
+				else {
 					frequency_list.push_back(i);
-					description_list.push_back("R: " + to_string_dec_uint(entry.frequency_a) + ">" + to_string_dec_uint(entry.frequency_b));
-																						}
+		description_list.push_back("R: " + to_string_dec_uint(entry.frequency_a) + ">" + to_string_dec_uint(entry.frequency_b) + ":S:" + to_string_dec_uint(step));
```

Comment:
Here, for a range (which may be hundreds of frequencies)  you are just placing the same string, over and over, inflating memory with it. Maybe you can only place the description on the first item, then just push an empty string (or a token, like "#") and later, when displaying the descriptions while scanning, when you find the empty (or token) string, you just don't update the txt field ? This would potentially save hundreds of ram bytes ? 

I will try this approach on my repository.